core/storage/suggestion/gae_models.py

```diff
@@ -693,6 +693,15 @@ def get_in_review_translation_suggestions_with_exp_ids_by_offset(
             language_codes: list(str). The list of language codes.
             exp_ids: list(str). Exploration IDs matching the target ID of the
                 translation suggestions.
+            Maximum number of entities to be returned: Note that the None case
+                is needed because we sometimes need to fetch all translation
+                suggestions so that they can be sorted client-side based on
+                specific criteria, and the datastore does not efficiently
+                support sorting based on complex client-side criteria. So, we
+                need to fetch all matching entities from the datastore and
+                allow the frontend to perform the sorting as needed.
+                Even in this case, though, the maximum number of entities
+                that can be returned by this query is 500.
```

Comment:
Wait, why? This needs explanation.

Reply:
I thought you wanted the note to be what you mentioned here in this thread https://github.com/oppia/oppia/pull/18664#discussion_r1278509407. I have removed this part now, thanks.